ros2action/ros2action/api/__init__.py

```diff
@@ -139,3 +142,23 @@ def __call__(self, prefix, parsed_args, **kwargs):
                 if n == action_name:
                     return t
         return []
+
+
+class ActionGoalPrototypeCompleter:
+    """Callable returning an action goal prototype."""
+
+    def __init__(self, *, action_type_key=None):
+        self.action_type_key = action_type_key
+
+    def __call__(self, prefix, parsed_args, **kwargs):
+        full_action_type = getattr(parsed_args, self.action_type_key)
+        try:
+            parts = full_action_type.split('/')
+            package_name = parts[0]
+            interface_type = parts[1]
+            action_type = parts[-1]
```

Comment:
`action_tutorials/Fibonacci` and `action_tutorials/action/Fibonacci` should be both supported (as this is `ros2 action` command, in the first case `action` is implicit).
Prefer checking for `len(parts)` than using a `try` `except` block.

Reply:
Updated this chunk on the model of  [`send_goal`](https://github.com/ros2/ros2cli/blob/master/ros2action/ros2action/verb/send_goal.py#L84).